crates/luminol-graphics/src/sprite/shader.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2023 Lily Lyons
+//
+// This file is part of Luminol.
+//
+// Luminol is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Luminol is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Luminol.  If not, see <http://www.gnu.org/licenses/>.
+
+fn create_shader(
+    render_state: &egui_wgpu::RenderState,
+    bind_group_layouts: &crate::BindGroupLayouts,
+    target: wgpu::BlendState,
+) -> wgpu::RenderPipeline {
+    let use_push_constants = render_state
+        .device
+        .features()
+        .contains(wgpu::Features::PUSH_CONSTANTS);
```

Comment:
If use_push_constants is going to be calculated from inside of luminol-graphics then ideally all calculations of this variable should be moved to luminol-graphics for consistency. Currently it's also being calculated from luminol-components.